numpy/core/src/umath/ufunc_object.c

```diff
@@ -5064,6 +5068,22 @@ ufunc_repr(PyUFuncObject *ufunc)
     return PyUString_FromFormat("<ufunc '%s'>", ufunc->name);
 }
 
+static int
+ufunc_traverse(PyObject *self, visitproc visit, void *arg)
+{
+    PyUFuncObject *ufunc = (PyUFuncObject*)self;
+    Py_VISIT(ufunc->obj);
+    return 0;
+}
+
+static int
+ufunc_clear(PyObject *self)
+{
+    PyUFuncObject *ufunc = (PyUFuncObject*)self;
+    Py_CLEAR(ufunc->obj);
+    return 0;
+}
```

Comment:
I think this is unnecessary - ufuncs are immutable, so it should not be possible for them to be in reference cycles with other immutable objects.

Furthermore, if you remove this then you can use the presence of `obj` in tp_alloc to decide whether the GC needs to untrack the object.